pyomo/contrib/pyros/util.py

```diff
@@ -0,0 +1,683 @@
+    timing_data_obj[code_block_name] = prev_time + elapsed_time
+
+
+def get_main_elapsed_time(timing_data_obj):
+    """Returns the time since entering the main `time_code` context"""
+    current_time = timeit.default_timer()
+    try:
+        return current_time - timing_data_obj.main_timer_start_time
+    except AttributeError as e:
+        if 'main_timer_start_time' in str(e):
+           raise AttributeError(
+                "You need to be in a 'time_code' context to use `get_main_elapsed_time()`."
+            )
+
+def a_logger(str_or_logger):
+    """Returns a logger when passed either a logger name or logger object."""
+    if isinstance(str_or_logger, logging.Logger):
+        return str_or_logger
+    else:
+        return logging.getLogger(str_or_logger)
+
+def ValidEnum(enum_class):
+    '''
+    Python 3 dependent format string
+    '''
+    def fcn(obj):
+        if obj not in enum_class:
+            raise ValueError("Expected an {0} object, "
+                             "instead recieved {1}".format(enum_class.__name__, obj.__class__.__name__))
+        return obj
+    return fcn
+
+class grcsTerminationCondition(Enum):
+    '''
+    Enum class to describe termination conditions of the grcs algorithm
+    robust_optimal: the grcs algorithm returned with a robust_optimal solution under normal conditions
+    max_iter: the grcs algorithm could not identify a robust optimal solution within the specified max_iter.
+              Consider increasing the max_iter config param.
+    error: there was an error in the grcs solution procedure. Check the log file.
+    '''
+    robust_feasible = 0
+    robust_optimal = 1
+    robust_infeasible = 2
+    max_iter = 3
+    subsolver_error = 4
+    time_out = 5
+
+
+class SeparationStrategy(Enum):
+    all_violations = auto()
+    max_violation = auto()
+
+
+class SolveMethod(Enum):
+    local_solve = auto()
+    global_solve = auto()
+
+
+class ObjectiveType(Enum):
+    worst_case = auto()
+    nominal = auto()
+
+def model_is_valid(model):
+    '''
+    Possibilities:
+    Deterministic model has a single objective
+    Deterministic model has no objective
+    Deterministic model has multiple objectives
+    :param model: the deterministic model
+    :return: True if it satisfies certain properties, else False.
+    '''
+    objectives = list(model.component_data_objects(Objective))
+    for o in objectives:
+        o.deactivate()
+    if len(objectives) == 1:
+        '''
+        Ensure objective is a minimization. If not, change the sense.
+        '''
+        obj = objectives[0]
+
+        if obj.sense is not minimize:
+            sympy_obj = sympyify_expression(-obj.expr)
+            # Use sympy to distribute the negation so the method for determining first/second stage costs is valid
+            min_obj = Objective(expr=sympy2pyomo_expression(sympy_obj[1].simplify(), sympy_obj[0]))
+            model.del_component(obj)
+            model.add_component("obj",min_obj)
```

Comment:
I think it may be safer in the long term to delete `obj`  because in the subproblems there are objective activation and deactivation steps and I don't want the original objective to accidentally get picked up somewhere in the algorithm. I also think it is okay to do because we are working on our `working_model` here, so it wont effect the user's model instance.

I will incorporate the `unique_component_name` suggestion